include/envoy/stream_info/filter_state.h

```diff
@@ -29,17 +30,23 @@ class FilterState {
   /**
    * @param data_name the name of the data being set.
    * @param data an owning pointer to the data to be stored.
-   * Note that it is an error to call setData() twice with the same data_name; this is to
-   * enforce a single authoritative source for each piece of data stored in FilterState.
+   * @param mutable indicates whether the object is mutable or not.
+   * Defaults to false.
+   *
+   * Note that it is an error to call setData() twice with the same
+   * data_name, if the existing object is immutable. This is to enforce a
+   * single authoritative source for each piece of immutable data stored in
+   * FilterState.
    */
-  virtual void setData(absl::string_view data_name, std::unique_ptr<Object>&& data) PURE;
+  virtual void setData(absl::string_view data_name, std::unique_ptr<Object>&& data,
```

Comment:
no default argument in virtual method.
https://google.github.io/styleguide/cppguide.html#Default_Arguments